connect/src/com/telenor/mobileconnect/ui/OperatorSelectionActivity.java

```diff
@@ -0,0 +1,108 @@
+
+    static final String MCC_MNC_PARAM_NAME = "mcc_mnc";
+    static final String MCC_MNC_PARAM_SEP = "_";
+    static final String SUBSCRIBER_ID_QUERY_PARAM_NAME = "subscriber_id";
+
+    @SuppressLint("MissingSuperCall")
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState, OperatorSelectionWebFragment.class);
+    }
+
+    @Override
+    public void onSuccess(Object successData) {
+
+        Uri uri = (Uri) successData;
+        String[] mccAndMnc = uri.getQueryParameter(MCC_MNC_PARAM_NAME).split(MCC_MNC_PARAM_SEP);
+
+        @SuppressWarnings("unchecked")
+        Map<String, String> parameters = (Map<String, String>) this
+                .getIntent()
+                .getExtras()
+                .get(ConnectUtils.PARAM_SNAPHOT_EXTRA);
+        @SuppressWarnings("unchecked")
+        List<String> uiLocales = (List<String>) this
+                .getIntent()
+                .getExtras()
+                .get(ConnectUtils.UI_LOCALES_SNAPHOT_EXTRA);
+        final int requestCode = this
+                .getIntent()
+                .getExtras()
+                .getInt(ConnectUtils.REQUEST_CODE_EXTRA);
+        final int customLoadingLayout = this
+                .getIntent()
+                .getExtras()
+                .getInt(ConnectUtils.CUSTOM_LOADING_SCREEN_EXTRA, ConnectSdk.NO_CUSTOM_LAYOUT);
+        MobileConnectSdkProfile sdkProfile = (MobileConnectSdkProfile) ConnectSdk.getSdkProfile();
+        sdkProfile.onStartAuthorization(
+                parameters,
+                uiLocales,
+                mccAndMnc[0],
+                mccAndMnc[1],
+                uri.getQueryParameter(SUBSCRIBER_ID_QUERY_PARAM_NAME),
+                new SdkProfile.OnStartAuthorizationCallback() {
+                    @Override
+                    public void onSuccess(Intent nextIntent) {
+                        if (!isConnectActivity(nextIntent)) {
+                            throw new RuntimeException("Wrong intent.");
+                        }
+                        Activity activity = OperatorSelectionActivity.this;
+                        if (customLoadingLayout != ConnectSdk.NO_CUSTOM_LAYOUT) {
+                            nextIntent.putExtra(
+                                    ConnectUtils.CUSTOM_LOADING_SCREEN_EXTRA,
+                                    customLoadingLayout);
+                        }
+                        activity.startActivityForResult(nextIntent, requestCode);
+                    }
+
+                    @Override
+                    public void onError() {
+                        cancelAndShowErrorMessage(OperatorSelectionActivity.this);
+                    }
+                });
+    }
+
+
+
+    @Override
+    public void onError(Object errorData) {
+        cancelAndShowErrorMessage(this);
+    }
+
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
```

Comment:
Ok, as I've done something quick for the Friday hack, just to show what I tried to say here:
https://github.com/telenordigital/connect-android-sdk/pull/57

A few things:

1. connect_activity is started right away. then it's responsibility to fetch:
    - in case of Connect Id - well-known config
    - in case of Mobile Connect - get mcc/mnc and then well-known, or if mcc/mnc fails, launch operator selection to get mcc/mnc and then well-known. I believe the code is 
easier to follow and it's less

2. Prefer using string literals for things that used only once

3. I've moved error handling to the client apps in the SigninActivity. I don't think the SDK needs to mess with the UI for the errors visualisation and that must be up to the app developer to decide.
   
4. I use anonymous classes as much as possible and put the callbacks definition as close as possible to where they are needed.